lib/signon/bootstrap.rb

```diff
@@ -6,34 +6,80 @@ module Signon
   module Bootstrap
     class NotRotatable < StandardError; end
 
-    def self.bootstrap_secrets(app_config:, aws:)
-      logger = Logger.new($stdout)
-      secrets_client = Aws::SecretsManager::Client.new(
-        region: aws.dig("region"),
-        credentials: aws.dig("credentials"),
+    def self.bootstrap_applications(applications:, signon:, secretsmanager:, logger: Logger.new($stdout))
+      applications.each do |app|
+        bootstrap_application(
+          app: app,
+          secretsmanager: secretsmanager,
+          signon: signon,
+          logger: logger,
+        )
+      end
+    end
+
+    def self.bootstrap_application(app:, secretsmanager:, signon:, logger:)
+      app_name = app.dig("name")
+      oauth_id_arn = app.dig("id_arn")
+      oauth_secret_arn = app.dig("secret_arn")
+
+      bootstrap_required = [oauth_id_arn, oauth_secret_arn].any? do |arn|
+        should_bootstrap(secretsmanager, arn)
+      end
+
+      unless bootstrap_required
+        logger.info "Secrets for #{app_name} already bootstrapped. Bailing out."
+        return
+      end
+
+      logger.info "Secret (#{app_name}) doesn't have secret values in SecretsManager. Creating them..."
+      secrets = find_or_create_app(signon, app)
+
+      logger.info "Application #{app_name} created or fetched. Putting creds in SecretsManager..."
+      secretsmanager.put_secret_value(
+        secret_id: oauth_id_arn,
+        secret_string: secrets.dig("oauth_id"),
```

Comment:
I believe we'd only want to rotate the Secret (the ID can stay the same, it's [not a secret](https://www.oauth.com/oauth2-servers/client-registration/client-id-secret/) - since it's used in URL queries etc).

For rotation we would probably want an app to have multiple secrets, but that wouldn't prevent us combining these into one secet.

gds-sso (the gem we use for OAuth between apps) [expects two environment variables](https://github.com/alphagov/gds-sso/blob/master/lib/gds-sso/config.rb#L13-L19). These could potentially be combined into one secret - but it's probably out of scope for this particular task - I've added https://trello.com/c/uBqmrWBN/312-combine-gdsssooauthid-and-gdsssooauthsecret-into-one-secret to follow up, if that's ok?

Reply:
Sure no probs - no need to worry about it in this PR. Sounds like we should ultimately be storing them as one secret with `{oauth_id: foo, oauth_secret: bar}` though.

Reply:
I'm assuming that if an app has multiple secrets then each of those would have a separate oauth_id? If that's not the case then I've been talking nonsense all along and you should ignore me 😅

Reply:
I think whether an app has multiple IDs will be up to us - we'll need to implement it from scratch, since the [Doorkeeper gem](https://github.com/doorkeeper-gem/doorkeeper) we use for OAuth doesn't support rotation of creds :(